contrib/epee/include/string_tools.h

```diff
@@ -59,6 +59,16 @@
 #pragma comment (lib, "Rpcrt4.lib")
 #endif
 
+namespace
+{
+	static inline uint8_t atolower(uint8_t c)
+	{
+		if (c >= 'A' && c <= 'Z')
+			c |= 32;
+		return c;
+	}
+}
+
```

Comment:
Just curious, why not just use `std::tolower()`?

Reply:
It was slower when I tried IIRC. Memory's hazy. If you time it to be faster, maybe it can change.

Reply:
`std::tolower` has to lookup the current locale setting, which involves the usage of a thread-safe global variable. This will mess with optimizations in a tight loop quite a bit - the code cannot be inlined AND involves an indirect jump every time.